src/Context/DataProvider.js

```diff
@@ -0,0 +1,216 @@
+import { DataContext } from "./DataContext";
+import { useState, useReducer } from "react";
+import * as Actions from "./ActionTypes";
+import { products } from "./Products";
+
+export const DataProvider = ({ children }) => {
+  const [toast, setToast] = useState(false);
+  const [toastMessage, setToastMessage] = useState("");
+
+  const dispatchFunction = (state, action) => {
+    switch (action.type) {
+      case Actions.ADD_TO_WISH:
+        return {
+          ...state,
+          wishProducts: addToWish(
+            action.payload,
+            state.wishProducts,
+            state.products
+          ),
+        };
+      case Actions.REMOVE_FROM_WISH:
+        return {
+          ...state,
+          wishProducts: removeFromWish(action.payload, state.wishProducts),
+        };
+      case Actions.MOVE_TO_CART_FROM_WISH:
+        return {
+          ...state,
+          cartProducts: moveToCart(
+            action.payload.id,
+            action.payload.count,
+            state.products,
+            state.cartProducts
+          ),
+          wishProducts: removeFromWish(action.payload.id, state.wishProducts),
+        };
+      case Actions.MOVE_TO_WISH_FROM_CART:
+        return {
+          ...state,
+          wishProducts: moveToWish(
+            action.payload.id,
+            action.payload.count,
+            state.products,
+            state.wishProducts
+          ),
+          cartProducts: removeFromCart(action.payload.id, state.cartProducts),
+        };
+      case Actions.ADD_TO_CART:
+        return {
+          ...state,
+          cartProducts: addToCart(
+            action.payload,
+            state.products,
+            state.cartProducts
+          ),
+        };
+      case Actions.REMOVE_FROM_CART:
+        return {
+          ...state,
+          cartProducts: removeFromCart(action.payload, state.cartProducts),
+        };
+      case Actions.INCREASE_ITEM_IN_WISH:
```

Comment:
wish everywhere shud be renamed to wishlist. Please go through all the action cases, they can be refactored. like increase and decrease qty can be combined to make a single case UPDATE_QUANTITY
Functions are well written, naming is well done.

Reply:
Yeah Even I thought of renaming wish to Wishlist.
And Ill move the reducer function and other function to seperate file.
Im not sure modifying the Increase and decrease to update quantity but ill try doing .